src/lib/__tests__/insights-client-test.ts

```diff
@@ -0,0 +1,231 @@
+import { withInsights, inferInsightsPayload } from '../insights';
+
+const connectHits = (renderFn: any, unmountFn: any) => (widgetParams = {}) => ({
+  init() {},
+  render({ results, instantSearchInstance }) {
+    const hits = results.hits;
+    renderFn({ hits, results, instantSearchInstance, widgetParams }, false);
+  },
+  dispose() {
+    unmountFn();
+  },
+});
+
+describe('withInsights', () => {
+  let instantSearchInstance: any;
+  let hits: any;
+  let results: any;
+  let widgetParams: any;
+  let widget: any;
+  let renderFn: any;
+
+  beforeEach(() => {
+    instantSearchInstance = {
+      insightsClient: jest.fn(),
+    };
+    hits = [
+      { objectID: '1', __position: 9 },
+      { objectID: '2', __position: 10 },
+      { objectID: '3', __position: 11 },
+      { objectID: '4', __position: 12 },
+    ];
+    results = {
+      index: 'theIndex',
+      queryID: 'theQueryID',
+      hits,
+    };
+    widgetParams = { transformItems: jest.fn() };
+
+    renderFn = jest.fn();
+    const unmountFn = jest.fn();
+
+    const connectHitWithInsights = withInsights(connectHits as any);
+    widget = connectHitWithInsights(renderFn, unmountFn)(widgetParams);
+    widget.render({ results, instantSearchInstance });
+  });
+
+  describe('when applied on connectHits', () => {
+    it('should call the passed renderFn', () => {
+      expect(renderFn).toHaveBeenCalledTimes(1);
+    });
+
+    it('should not remove any renderProps passed by connectHits', () => {
+      const [renderProps] = renderFn.mock.calls[0];
+      expect(renderProps).toEqual(
+        expect.objectContaining({
+          instantSearchInstance,
+          results,
+          widgetParams,
+        })
+      );
+    });
+
+    it('should expose the insights client wrapper to renderOptions if passed to instantSearchInstance', () => {
+      const [renderProps] = renderFn.mock.calls[0];
+      expect(renderProps).toHaveProperty('insights');
+      expect(renderProps.insights).toBeInstanceOf(Function);
+    });
+
+    it('should not expose the insights client wrapper to renderOptions if not passed to instantSearchInstance', () => {
+      delete instantSearchInstance.insightsClient;
+      const [renderProps] = renderFn.mock.calls[0];
+      expect(renderProps).toHaveProperty('insights');
+      expect(renderProps.insights).toBeInstanceOf(Function);
+    });
+  });
+
+  describe('exposed insights client wrapper', () => {
```

Comment:
Some tests can be grouped together.

Reply:
could you tell which?
IMHO it's best that each tests is no longer than a couple of lines (or statements).